formats/compact-refract.md

```diff
@@ -0,0 +1,96 @@
+# Compact Refract
+
+Compact Refract is a serialization format Refract for the purpose of removing a
+lot of the object keys that are repeated throughout the full serialization of
+Refract, as seen in the Refract specification. It also allows for expressing
+structures in a tuple, which resembles other formats like XML or Lisp.
+
+## Dependencies
+
+- [Refract Base
+Specification](https://github.com/refractproject/refract-spec/blob/master/refract-spec.md)
+  - Only Include
+    - Element Pointer
+
+## Data Structures
+
+### Compact Element (array)
+
+The Compact Element is a tuple where each item has a specific meaning. The
+first item is the element name, the second is the meta attribute section, the
+third is the attribute section, and the fourth is the content section.
+
+#### Members
+
+- (string, required) - Name of the element
+- (Meta Attributes, required) - Meta attributes of the element instance.
+- (object, required) - Attributes of the element instance
+- (enum, required) - Element content with any of the following types
+  - (null)
+  - (string)
+  - (number)
+  - (boolean)
+  - (array)
+  - (object)
+  - (Compact Element)
+  - (array[Compact Element])
+
+## Meta Attributes
+
+The Meta Attributes are similar to those defined by the base specification,
+though the only difference is that Link Elements are of a different structure.
+
+### Members
+
+- `id` - Unique Identifier, MUST be unique throughout the document
+- `ref` (Element Pointer) - Pointer to referenced element or type
+- `classes` (array[string]) - Array of classifications for given element
+- `title` (string) - Human-readable title of element
+- `description` (string) - Human-readable description of element
+- `links` (array[Link Element]) - Meta links for a given element
+
+## Link Element (array)
+
+Allow for hyperlinking within a Refract document. Note that this Link Element
+is different than the Link Element in the base specification.
+
+### Members
+
+- link (string, fixed)
+- (Meta Attributes)
+- (object)
+  - relation (string) - Link relation type as specified in RFC 5988.
+  - href (string) - The URI for the given link
+
+## Example
+
+In the base specification, serialization looks something like this for JSON:
+
+```json
+{
+  "element": "foo",
+  "content": "bar"
+}
+```
+
+In Compact Refract, this same example looks like this:
+
+```json
+["foo", {}, {}, "bar"]
+```
+
+In this example, meta attributes and attributes are both required, even if the
+objects are empty. This is because this format relies on the structure of the
+array in order to determine values.
+
+## Pros and Cons
+
+The benefits as mentioned are that this is a more concise format. In some
+cases, it may even be easier to write, or convert from other formats such as
+XML or Lisp.
+
+One downside of this format is that there is no simple way to distinguish a
```

Comment:
@danielgtaylor I have added a note that defines how to handle this a little differently. Please check out the latest commit.

Basically, it says if an array is found that is ambiguous, it SHOULD be treated as a normal array and not a Compact Refract element. It is up to the user whether to convert that or handle in a special way according to their tooling or domain specification.

Thoughts?